exercises/07-normalizr/src/modules/entities/entities-slice.ts

```diff
@@ -0,0 +1,41 @@
+import { createSlice, PayloadAction, CaseReducer } from '@reduxjs/toolkit'
+
+import mergeWith from 'lodash/mergeWith'
+import isArray from 'lodash/isArray'
+
+import { UserEntities, NormalizedUserEntities } from './entities-types'
+import { usersActions } from 'modules/users/users-slice'
+
+const initSate: UserEntities = {
+  skills: {},
+  userSkills: {},
+  users: {},
+}
+
+const replaceArraysCustomizer = (objValue: unknown, srcValue: unknown) => {
+  // if merging two arrays, replace original value with new one
+  if (isArray(objValue) && isArray(srcValue)) {
+    return srcValue
+  }
+
+  // otherwise merge as expected
+  return undefined
+}
+
+const updateEntities: CaseReducer<
+  UserEntities,
+  PayloadAction<NormalizedUserEntities>
```

Comment:
I am not sure about this typing. I mean now you are effectively tying `NormalizedUserEntities` (concrete) with `updateEntities` (abstract).

Reply:
Maybe it needs just a better naming because `NormalizedUserEntities` is an object of all entities in the app.

Reply:
Rewritten in e656a05

Reply:
Currently the `NormalizedUserEntities` is an object of only `UserEntities`. It does describe the return type of `normalize` call when called with fetched users and `Schema.users` schema. I imagined that when other unrelated entities are added, e.g. towns, they would have separate `TownEntities` and `NormalizedTownEntities`.

Should have I made it different?

Reply:
I think it's fine now. I imagine that once you would extend the app by `TownEntities`, the `EntitiesState` will look like this

```ts
interface EntitiesState extends UserEntities, TownEntities {}
```

Reply:
Exactly :)

Reply:
Rewritten with type 539f0b9